### PR TITLE
Stick to Django support's roadmap

### PR DESCRIPTION
**Pull request by anapaulagomes**
Resolves https://github.com/vandersonmota/model_mommy/issues/365.

Plan:

- [x] Drop support to Python 2
- [x] Stick to [Django's roadmap](https://www.djangoproject.com/download/)

### CODE REVIEW

**Comment by anapaulagomes** [comment]
@berinhard I need your input in here to make sure I'm not forgetting anything. 😅

File: docs/source/index.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@ Examples of contributions include:
 Compatibility
 =============
 
-model_mommy supports Django >= 1.8
+model_mommy supports Django >= 1.11
```
Comment (amureki) [inline docs/source/index.rst]: Shall we add a note about python 3 only?

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
-django>=1.8.0
+django>=1.11.0
 six
```
Comment (amureki) [inline requirements.txt]: Do we still need six here?

File: tox.ini [hunk 3]
```diff
@@ -1,26 +1,18 @@
 [tox]
 envlist =
-    py27-django{18,19,110,111}-{postgresql,sqlite}
-    py35-django{18,19,110,111,20}-{postgresql,sqlite}
+    py35-django{111,20}-{postgresql,sqlite}
```
Comment (amureki) [inline tox.ini]: what about adding Django 2.1 as well?

File: tox.ini [hunk 4]
```diff
@@ -1,26 +1,18 @@
...
   PYTHONPATH = {toxinidir}
 basepython =
-    py27: python2.7
     py35: python3.5
     py36: python3.6
 deps =
-    py26: Pillow>=2.3.1,<4.0.0
-    py{27,34,35}: Pillow>=2.3.1
-    mock==1.0.1
+    py{34,35}: Pillow>=2.3.1
```
Comment (amureki) [inline tox.ini]: Did you mean 35,36 here?

File: requirements.txt (hunk 2 above)
Comment (anapaulagomes) [reply requirements.txt]: Yes, we do. We're using functions such as `text_type`, `string_types`, and `advance_iterator`.

File: tox.ini (hunk 3 above)
Comment (anapaulagomes) [reply tox.ini]: I'm on it. :) Actually, I'm adding support to Python3.7 too.

File: tox.ini (hunk 4 above)
Comment (anapaulagomes) [reply tox.ini]: Actually, I ran tox without this dependency (which is not in `requirements.txt` or `dev_requeriments.txt`) and the tests passed. @berinhard do we need it for some reason?

**Comment by anapaulagomes** [comment]
> What is your idea on this, do you want it to be model_mommy 2.0.0?
Shall we resolve as many issues as we can and release updates before we do a breaking release?

Yes, I think would be nice to publish it as `model_mommy 2.0.0`. Maybe we can focus on fix/release #377 and #338, and also adding a warning about the supported versions. My concern about resolving many issues before it is adding more code with old support.

**Comment by berinhard** [comment]
Amazing @anapaulagomes! I think there is still some more code clean up to be done, but it can wait. Look very good for me :+1:

**Pull request merged**